src/java/org/apache/cassandra/service/accord/AccordSafeCommandStore.java

```diff
@@ -174,19 +187,34 @@ private <O> O mapReduceForKey(Routables<?, ?> keysOrRanges, Ranges slice, BiFunc
                     if (accumulate.equals(terminalValue))
                         return accumulate;
                 }
-                break;
+            }
+            break;
             case Range:
-                // TODO (required): implement
-                throw new UnsupportedOperationException();
+            {
+                // TODO (correctness): if a range is used, then the range must exist in the PreLoadContext, else the commandsForKeys won't be in-sync... can this be detected?
```

Comment:
It can, it's a safe command store so it still has the preload context?

Reply:
added defensive check